packages/expect/src/jestMatchersObject.ts

```diff
@@ -6,24 +6,22 @@
  *
  */
 
+import type {Expect} from '@jest/types';
 import {AsymmetricMatcher} from './asymmetricMatchers';
-import type {
-  Expect,
-  MatcherState,
-  MatchersObject,
-  SyncExpectationResult,
-} from './types';
 
 // Global matchers object holds the list of available matchers and
 // the state, that can hold matcher specific values that change over time.
 const JEST_MATCHERS_OBJECT = Symbol.for('$$jest-matchers-object');
 
-// Notes a built-in/internal Jest matcher.
-// Jest may override the stack trace of Errors thrown by internal matchers.
-export const INTERNAL_MATCHER_FLAG = Symbol.for('$$jest-internal-matcher');
+// Expect may override the stack trace of Errors thrown by built-in matchers.
+export const BUILD_IN_MATCHER_FLAG = Symbol.for('$$build-in-matcher');
+
+export type BuildInRawMatcherFn = Expect.RawMatcherFn & {
+  [BUILD_IN_MATCHER_FLAG]?: boolean;
+};
```

Comment:
Should not be optional, because this prop will be added to each `matcher`. At the same time, in code it works only as optional prop. Hm.. Feels like it does not belong to this type.